src/devices/Units/Pressure.cs

```diff
@@ -120,51 +120,110 @@ public static Pressure FromMillimeterOfMercury(double value)
             return new Pressure(value / MillimeterOfMercuryRatio);
         }
 
+        /// <inheritdoc cref="IComparable{T}.CompareTo" />
+        public int CompareTo(Pressure other)
+        {
+            return _pascal.CompareTo(other._pascal);
+        }
+
         /// <summary>
-        /// Creates a string representation of this object, in hPa.
+        /// Returns the string representation of this pressure, in hPa
         /// </summary>
+        /// <returns>String representation of this pressure</returns>
         public override string ToString()
         {
-            return string.Format(CultureInfo.CurrentCulture, "{0:F2}hPa", Hectopascal);
+            return String.Format(CultureInfo.CurrentCulture, "{0:F1} hPa", Hectopascal);
         }
 
         /// <summary>
-        /// Equality member
+        /// Returns the string representation of this pressure, with the given format string and using the current culture.
+        /// For valid formatting arguments, see <see cref="ToString(string, IFormatProvider)"/>
         /// </summary>
-        public bool Equals(Pressure other)
+        /// <param name="formatArgs">Format string</param>
+        /// <returns>String representation</returns>
+        public string ToString(string formatArgs)
         {
-            return _pascal.Equals(other._pascal);
+            return ToString(formatArgs, CultureInfo.CurrentCulture);
         }
 
         /// <summary>
-        /// Equality member
+        /// Returns the string representation of this pressure, with the given format string and using the given culture.
+        /// Valid format specifiers are:
+        /// PA: Pascal
+        /// MBAR: Millibar
+        /// KPA: Kilopascal
+        /// HPA: Hectopascal
+        /// INHG: Inch of mercury
+        /// MMHG: Millimeter of mercury
+        /// An extra number defines the number of decimal digits to use (default 1)
+        /// <example>
+        /// <code>
+        /// var s = p.ToString("PA2"); -> "101325.01 Pa"
+        /// var s = p.ToString("HPA2"); --> "1013.25 hPa"
+        /// </code>
+        /// </example>
         /// </summary>
-        public override bool Equals(object obj)
+        /// <param name="formatArgs">Format string</param>
+        /// <param name="culture">Culture to use. Affects the format of the number.</param>
+        /// <returns>String representation</returns>
+        public string ToString(string formatArgs, IFormatProvider culture)
```

Comment:
I'm assuming those are made up formats and not officially documented anywhere, if so please create an issue about them (if there are similar for temperature also mention them) since we need to make sure everyone agrees on that before we ship that - it will be super breaking to change that in the future so let's make sure we get those right.

I think it's fine not to block PR on that (just please make sure you check case insensitive (pass comparer)) and that you create an issue

Reply:
note: there is a chance we will remove those formatters if we don't agree on them in the issue by the time we start shipping preview.

Reply:
I've created #1047 on this, since it was originally my idea.